integration/internal/api/volume/volume.go

```diff
@@ -0,0 +1,61 @@
+package volume
+
+import (
+	"context"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	volumetypes "github.com/docker/docker/api/types/volume"
+	"github.com/docker/docker/client"
+)
+
+// Create creates a volume using the named driver with the specified options
+func Create(client client.APIClient, driver string, opts map[string]string) (string, error) {
+	volReq := volumetypes.VolumesCreateBody{
+		Driver:     driver,
+		DriverOpts: opts,
+		Name:       "",
+	}
+
+	ctx := context.Background()
+	vol, err := client.VolumeCreate(ctx, volReq)
+	if err != nil {
+		return "", err
+	}
+	return vol.Name, nil
+}
+
+// Rm removes the volume named
+func Rm(client client.APIClient, name string) error {
+	ctx := context.Background()
+	return client.VolumeRemove(ctx, name, false)
+}
+
+// Ls lists the volumes available
+func Ls(client client.APIClient) ([]string, error) {
+	ctx := context.Background()
+	volumes, err := client.VolumeList(ctx, filters.Args{})
+	if err != nil {
+		return []string{}, err
+	}
+
+	names := []string{}
+	for _, volume := range volumes.Volumes {
```

Comment:
We should not be maintaining an abstraction over the APIClient for tests. Please remove this file. The only logic that really needs to be a new function is `namesFromVolumes()` which can be an unexported helper next to the one place it is used.

Reply:
While it may be desirable in the long term to not have any abstraction over `APIClient` for tests, this design makes porting CLI tests much easier and reduces incidental complexity significantly. To maintain correctness, I recommend not attempting to simultaneously port tests and radically change them or instill new best practices.

Across existing tests, similar functionality will be required:

```
$ git grep "volume\", \"ls\")" integration-cli/ | wc -l
      13
```

```
$ git grep "volume\", \"ls" integration-cli/ | wc -l
      42
```

As this functionality is in `integration/internal`, golang's magic export scoping rules for `internal` will prevent users outside of this repository.